xen/arch/x86/boot/slaunch_early.c

```diff
@@ -127,5 +165,6 @@ uint32_t __stdcall txt_early_tests(uint32_t load_base_addr,
 
     verify_pmr_ranges(os_mle, os_sinit, load_base_addr, tgt_base_addr, size);
 
-    return os_mle->boot_params_addr;
+    result->mbi_pa = os_mle->boot_params_addr;
+    result->slaunch_param = slaunch_param;
```

Comment:
Would returning `os_mle->slrt` here simplify later code? That way getting SLRT pointer would be the same for Intel and AMD. If that isn't a trivial change I'm OK with leaving it for further improvement, it isn't part of SKINIT support.

Reply:
Not that it's a drastic simplification, but it's 26 less lines and, more importantly, more consistent behaviour between Intel and AMD.  https://github.com/TrenchBoot/xen/compare/6c44eb9bcc4f73b6b98e6a6cfd50c0994c85580c..0eac7a817733f1f1d3a7ab853e49ad47473d0396